tests/eval_tests/tests.py

```diff
@@ -353,6 +345,16 @@ def main():
         client_tool_queries = os.path.join(os.path.dirname(os.path.realpath(__file__)),
                                            "queries/", "client_tool_queries.json")
 
+    tool_list = []
+    tool_name_set = set()
```

Comment:
Why is this a set and not another list? Why do you need to maintain the tool list and the tool names?